modules/ting_smart_search/ting_smart_search.module

```diff
@@ -456,29 +475,34 @@ function ting_smart_search_get_autodata_path() {
  */
 function ting_smart_search_get_autodata_from_webtrekk() {
   $url = variable_get('ting_smart_search_master_server', 'https://staging.randersbib.dk/data/autodata.txt');
-  $path = ting_smart_search_get_autodata_path();
-  $temp_path = $path . '.temp';
+  $path = file_default_scheme() . '://' . variable_get('ting_smart_search_autodata_path', 'smartsearchdata') . DIRECTORY_SEPARATOR . 'autodata.txt';
+
   try {
     $directory = dirname($path);
     if (!file_prepare_directory($directory, FILE_CREATE_DIRECTORY | FILE_MODIFY_PERMISSIONS)) {
       watchdog('ting_smart_search', 'Failed to create directory: %directory', array('%directory' => $directory), WATCHDOG_ERROR);
       return FALSE;
     }
-    if (system_retrieve_file($url, $temp_path, FALSE, FILE_EXISTS_REPLACE)) {
-      $data = file_get_contents($temp_path);
+    if (system_retrieve_file($url, $path, FALSE, FILE_EXISTS_REPLACE)) {
+      $data = file_get_contents($path);
       if (isset($data)) {
         $output = unserialize($data);
       }
       if (isset($output) && !empty($output) && is_array($output)) {
-
-        $val = var_export($output, TRUE);
-        // In order to get maximum performance the data is exported into a php
-        // array which will be cached by opcode cache. This is based on the method
-        // described here: https://medium.com/@dylanwenzlau/
-        // 500x-faster-caching-than-redis-memcache-apc-in-php-hhvm-dcd26e8447ad
-        // Build the destination folder tree if it doesn't already exist.
-        // LOCK_EX was used in the original solution doesn't work, maybe a windows issue.
-        file_put_contents($path, '<?php $ting_smart_search_autodata = ' . $val . ';');
+        db_drop_table('ting_smart_search_auto_data');
+        db_create_table('ting_smart_search_auto_data', ting_smart_search_db_table());
+
+        foreach ($output as $searchkey => $pids) {
+          $material_ids = implode(',', array_keys($pids));
+          $f = $pids;
+          $query = db_insert('ting_smart_search_auto_data')->fields(
+                  [
+                      'searchkey' => $searchkey,
+                      'materialids' => $material_ids
+                  ]
+          );
+          $query->execute();
+        }
```

Comment:
How many search keys are you potentially inserting here? Maybe use multi insert query instead of separate transaction for each insert? It can give really bad performance if you're inserting thousands of rows in separate transaction.

Reply:
Has been fixed